sys/include/seqno.h

```diff
@@ -0,0 +1,321 @@
+/*
+ * Copyright (C) 2015 Cenk Gündoğan <cnkgndgn@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_seqno Serial Number Arithmetic
+ * @ingroup     sys
+ * @brief       Serial Number Arithmetic (RFC 1982)
+ * @{
+ *
+ * @file
+ * @brief       Serial Number Arithmetic (RFC 1982)
+ * @see         <a href="https://tools.ietf.org/html/rfc1982">
+ *                  RFC 1982 - Serial Number Arithmetic
+ *              </a>
+ *
+ * @author      Cenk Gündoğan <cnkgndgn@gmail.com>
+ */
+
+#ifndef SEQNO_H
+#define SEQNO_H
+
+#include <stdint.h>
+#include <errno.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define SEQNO_SERIAL_BITS(X)    (sizeof(X) * 8)
+#define SEQNO_BOUNDARY(X)       (1ULL << (SEQNO_SERIAL_BITS(X) - 1))
```

Comment:
Documentation for those defines would be nice, especially boundary is not clear to me (shouldn't this be `((1ULL << SEQNO_SERIAL_BITS(X)) - 1)`)